arangod/Aql/ClusterQuery.cpp

```diff
@@ -75,10 +68,14 @@ std::shared_ptr<ClusterQuery> ClusterQuery::create(
         : ClusterQuery{id, std::move(ctx), std::move(options)} {}
 
     ~MakeSharedQuery() final {
-      // Unregister this query from the query list, otherwise it's still
-      // accessible via this list while the query is being destructed,
+      try {
+        _traversers.clear();
+      } catch (...) {
+      }
```

Comment:
I think this should be kept in the ClusterQuery dtor as before. Even though TSan reported no related races (yet), I find it rather discomforting that we destroy the traversers of a query _while it might still be accessed_, because we are only waiting for the query to finish execution in `destroy`.

Reply:
Good point, will be done if build fail or in separate pr

Reply:
Done